test/built-ins/TypedArray/prototype/subarray/speciesctor-get-ctor-inherited.js

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.subarray
+description: get inherited constructor on SpeciesConstructor
+info: >
+  22.2.3.27 %TypedArray%.prototype.subarray( begin , end )
+
+  ...
+  17. Return ? TypedArraySpeciesCreate(O, argumentsList).
+
+  22.2.4.7 TypedArraySpeciesCreate ( exemplar, argumentList )
+
+  ...
+  3. Let constructor be ? SpeciesConstructor(exemplar, defaultConstructor).
+  ...
+
+  7.3.20 SpeciesConstructor ( O, defaultConstructor )
+
+  1. Assert: Type(O) is Object.
+  2. Let C be ? Get(O, "constructor").
+  3. If C is undefined, return defaultConstructor.
+  ...
+includes: [testTypedArray.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA([40, 41, 42, 43]);
+  var calls = 0;
+  var result;
+
+  Object.defineProperty(TA.prototype, "constructor", {
+    get: function() {
+      calls++;
+    }
+  });
+
+  result = sample.subarray(0);
+
+  assert.sameValue(calls, 1, "called custom ctor get accessor once");
+
+  assert.sameValue(
+    Object.getPrototypeOf(result),
+    Object.getPrototypeOf(sample),
+    "use defaultCtor on an undefined return - getPrototypeOf check"
+  );
+  assert.sameValue(
+    result.constructor,
+    undefined,
+    "used defaultCtor but still checks the inherited .constructor"
```

Comment:
I don't think I explained myself very well. Let me try again.

The confusing part for me was that the assertion is "proving" that
`result.constructor` is `undefind`, but there are many ways this could be true.
It could be that the instance has an "own" property whose value is `undefined`,
it could be that the instance has no such "own" property, but that its
prototype does, or it could be that a property with that name does not exist
anywhere in the prototype chain at all.

The assertion message helps explain what is happening, but I was suggesting
explicit assertions that would also verify that.

- An assertion using `hasOwnProperty` would clearly demonstrate that the value
  `undefined` is *not* coming directly from the `result` object itself.

    ```js
    assert.sameValue(Object.hasOwnProperty.call(result, "constructor"), false);
    ```

- An assertion that `calls` is now `2` would demonstrate that the previous
  assertion had triggered the invocation of the accessor property.

    ```js
    assert.sameValue(calls, 2);
    ```

  The newest assertion  does not demonstrate this because it re-sets the
  `calls` variable. So as written, it seems like an unrelated test for accessor
  properties. Asserting the value `2` *does* introduce a slight dependency
  between the assertions, and I can appreciate if you want to avoid that. You
  could avoid this by re-setting `calls` to `0` immediately before the
  assertion.

That said, we are definitely going too far into tiny details. I haven't
forgotten that the test is valid, all this discussion is just about a minor
readability improvement. I think it's fine to leave it as-is.

> That's the scope for speciesctor-get-ctor only.

Ah, of course.